### DISCUSSION TITLE
How to set kubernetes client request timeout during runtime

### DISCUSSION THREAD

**Discussion by AleksanderBrzozowski** [quarkusio/quarkus#31395] [Q&A] [answered]
Hi, I am having difficulties with configuring kubernetes client during runtime.
For example, I have three different environments, and one environment requires increased request timeout.

Since [kubernetes client properties](https://quarkus.io/guides/kubernetes-client#configuration-reference) are build time properties, they can be only overridden during build time.

How I can deal with that without specifying the highest value in properties during runtime? I want to have one docker image and possibility to override these properties when starting the container 🙂

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
I have opened https://github.com/quarkusio/quarkus/pull/31398 which will allow you to perform arbitrary customizations of the kubernetes config

**Reply by gsmet** [reply]
@geoand I had a look at the properties and it looks like some of them should be runtime. Is there any reason why they ended up all build time?

**Reply by geoand** [reply]
Yes, the reason is that the client is also used at build time to deploy an application

**Reply by AleksanderBrzozowski** [reply]
@geoand I saw the PR for this ticket, but it is not 100% clear to me how I should override the defaults in runtime.

I think that it is not possible through standard properties - does it mean that I need to provide `KubernetesConfigCustomizer` bean to override kubernetes client config?

**Reply by geoand** [reply]
Yes, that's the idea
